### ISSUE TITLE
Debug assertion: Logical error: 'No available columns' in Merge storage.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#18793]
https://clickhouse-test-reports.s3.yandex.net/18775/d77dfeaafb21d68ca319b1b549340435edd9c1b6/fuzzer/report.html#fail1

```
SELECT toDateTime(NULL)
FROM merge(currentDatabase(), 'test_local_1')
PREWHERE _table = 'test\0local_1test_local_1test\0local_1test_local_1'

[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.131064 [ 65 ] {b5c2d851-a002-444e-917f-e2f0b06eb6aa} <Fatal> : Logical error: 'No available columns'.
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.132001 [ 231 ] <Fatal> BaseDaemon: ########################################
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.132333 [ 231 ] <Fatal> BaseDaemon: (version 20.13.1.5603, build id: 5CB8F2FAA71C5CDF) (from thread 65) (query_id: b5c2d851-a002-444e-917f-e2f0b06eb6aa) Received signal Aborted (6)
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.132610 [ 231 ] <Fatal> BaseDaemon: 
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.132984 [ 231 ] <Fatal> BaseDaemon: Stack trace: 0x7fbf13099fb7 0x7fbf1309b921 0x1193af7c 0x1193b021 0x1abee84a 0x1b540037 0x1b541883 0x1b06d6f4 0x1b06c032 0x1b06cb1b 0x1b910e48 0x1b90f7ca 0x1b7a874f 0x1b077052 0x1b06ffef 0x1b06f4d6 0x1b38197a 0x1b38217d 0x1b5a67ed 0x1b5a50ca 0x1c060ac9 0x1c06bf78 0x1fbf573c 0x1fbf5f40 0x1fd342d3 0x1fd3120d 0x1fd30098 0x7fbf136576db
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.133394 [ 231 ] <Fatal> BaseDaemon: 4. /build/glibc-S7xCS9/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51: gsignal @ 0x3efb7 in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.27.so
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.133601 [ 231 ] <Fatal> BaseDaemon: 5. /build/glibc-S7xCS9/glibc-2.27/stdlib/abort.c:81: __GI_abort @ 0x40921 in /usr/lib/debug/lib/x86_64-linux-gnu/libc-2.27.so
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.134064 [ 231 ] <Fatal> BaseDaemon: 6. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:50: DB::handle_error_code(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1193af7c in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.134449 [ 231 ] <Fatal> BaseDaemon: 7. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:56: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x1193b021 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.170882 [ 231 ] <Fatal> BaseDaemon: 8. ./obj-x86_64-linux-gnu/../src/Interpreters/ExpressionActions.cpp:528: DB::ExpressionActions::getSmallestColumn(DB::NamesAndTypesList const&) @ 0x1abee84a in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.210544 [ 231 ] <Fatal> BaseDaemon: 9. ./obj-x86_64-linux-gnu/../src/Interpreters/TreeRewriter.cpp:630: DB::TreeRewriterResult::collectUsedColumns(std::__1::shared_ptr<DB::IAST> const&, bool) @ 0x1b540037 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.249778 [ 231 ] <Fatal> BaseDaemon: 10. ./obj-x86_64-linux-gnu/../src/Interpreters/TreeRewriter.cpp:799: DB::TreeRewriter::analyzeSelect(std::__1::shared_ptr<DB::IAST>&, DB::TreeRewriterResult&&, DB::SelectQueryOptions const&, std::__1::vector<DB::TableWithColumnNamesAndTypes, std::__1::allocator<DB::TableWithColumnNamesAndTypes> > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::TableJoin>) const @ 0x1b541883 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.286960 [ 231 ] <Fatal> BaseDaemon: 11. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:336: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&)::$_2::operator()(bool) const @ 0x1b06d6f4 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.324817 [ 231 ] <Fatal> BaseDaemon: 12. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:427: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) @ 0x1b06c032 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.362719 [ 231 ] <Fatal> BaseDaemon: 13. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:160: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, DB::SelectQueryOptions const&) @ 0x1b06cb1b in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.404525 [ 231 ] <Fatal> BaseDaemon: 14. ./obj-x86_64-linux-gnu/../src/Storages/StorageMerge.cpp:315: DB::StorageMerge::createSources(std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::QueryProcessingStage::Enum const&, unsigned long, DB::Block const&, std::__1::tuple<std::__1::shared_ptr<DB::IStorage>, std::__1::shared_ptr<DB::RWLockImpl::LockHolderImpl>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, std::__1::shared_ptr<DB::Context> const&, unsigned long, bool, bool) @ 0x1b910e48 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.446525 [ 231 ] <Fatal> BaseDaemon: 15. ./obj-x86_64-linux-gnu/../src/Storages/StorageMerge.cpp:227: DB::StorageMerge::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0x1b90f7ca in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.487554 [ 231 ] <Fatal> BaseDaemon: 16. ./obj-x86_64-linux-gnu/../src/Storages/IStorage.cpp:108: DB::IStorage::read(DB::QueryPlan&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0x1b7a874f in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.525393 [ 231 ] <Fatal> BaseDaemon: 17. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:1566: DB::InterpreterSelectQuery::executeFetchColumns(DB::QueryProcessingStage::Enum, DB::QueryPlan&, std::__1::shared_ptr<DB::PrewhereDAGInfo> const&, std::__1::unordered_set<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) @ 0x1b077052 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.563467 [ 231 ] <Fatal> BaseDaemon: 18. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:839: DB::InterpreterSelectQuery::executeImpl(DB::QueryPlan&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>) @ 0x1b06ffef in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.601545 [ 231 ] <Fatal> BaseDaemon: 19. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:487: DB::InterpreterSelectQuery::buildQueryPlan(DB::QueryPlan&) @ 0x1b06f4d6 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.639682 [ 231 ] <Fatal> BaseDaemon: 20. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:356: DB::InterpreterSelectWithUnionQuery::buildQueryPlan(DB::QueryPlan&) @ 0x1b38197a in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.678029 [ 231 ] <Fatal> BaseDaemon: 21. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:410: DB::InterpreterSelectWithUnionQuery::execute() @ 0x1b38217d in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.718404 [ 231 ] <Fatal> BaseDaemon: 22. ./obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:525: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x1b5a67ed in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.758676 [ 231 ] <Fatal> BaseDaemon: 23. ./obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:868: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0x1b5a50ca in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.805722 [ 231 ] <Fatal> BaseDaemon: 24. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:260: DB::TCPHandler::runImpl() @ 0x1c060ac9 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.852960 [ 231 ] <Fatal> BaseDaemon: 25. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1414: DB::TCPHandler::run() @ 0x1c06bf78 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.907648 [ 231 ] <Fatal> BaseDaemon: 26. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:43: Poco::Net::TCPServerConnection::start() @ 0x1fbf573c in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:22.961241 [ 231 ] <Fatal> BaseDaemon: 27. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerDispatcher.cpp:112: Poco::Net::TCPServerDispatcher::run() @ 0x1fbf5f40 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:23.016126 [ 231 ] <Fatal> BaseDaemon: 28. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/ThreadPool.cpp:199: Poco::PooledThread::run() @ 0x1fd342d3 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:23.071337 [ 231 ] <Fatal> BaseDaemon: 29. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread.cpp:56: Poco::(anonymous namespace)::RunnableHolder::run() @ 0x1fd3120d in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:23.126371 [ 231 ] <Fatal> BaseDaemon: 30. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread_POSIX.cpp:345: Poco::ThreadImpl::runnableEntry(void*) @ 0x1fd30098 in /workspace/clickhouse
[linux-ubuntu-16-04-xenial] 2021.01.06 05:18:23.126830 [ 231 ] <Fatal> BaseDaemon: 31. start_thread @ 0x76db in /lib/x86_64-linux-gnu/libpthread-2.27.so
```

**Comment by qoega** [comment]
``` sql
SELECT NULL FROM merge(currentDatabase(), 'test_local_1|test_local_2test_local_1|test_local_2') PREWHERE _table = ''
```

```
2021.03.23 10:19:47.307594 [ 46 ] {7088cf9c-1fc8-435e-860c-deb81aaccddd} <Fatal> : Logical error: 'No available columns'.
2021.03.23 10:19:47.309110 [ 183 ] {} <Fatal> BaseDaemon: ########################################
2021.03.23 10:19:47.309405 [ 183 ] {} <Fatal> BaseDaemon: (version 21.4.1.6317, build id: 7BCC6AB8C7C49828A7D44EC9F221D2791B423D4A) (from thread 46) (query_id: 7088cf9c-1fc8-435e-860c-deb81aaccddd) Received signal Aborted (6)
2021.03.23 10:19:47.309698 [ 183 ] {} <Fatal> BaseDaemon:
2021.03.23 10:19:47.310055 [ 183 ] {} <Fatal> BaseDaemon: Stack trace: 0x7fe7ebb2618b 0x7fe7ebb05859 0x1193d526 0x1193d621 0x1aaff82c 0x1b3e0232 0x1b3e1e46 0x1ac6c23f 0x1ac6a487 0x1ac6b5eb 0x1b763473 0x1b761c9d 0x1b68c3cf 0x1ac769cf 0x1ac6f41e 0x1ac6e456 0x1b22d42a 0x1b22db9d 0x1b4523e1 0x1b4508aa 0x1be33912 0x1be3ff28 0x1f8ea4bc 0x1f8eacc0 0x1fa27463 0x1fa242ed 0x1fa23178 0x7fe7ebcdb609 0x7fe7ebc02293
2021.03.23 10:19:47.310429 [ 183 ] {} <Fatal> BaseDaemon: 4. raise @ 0x4618b in /usr/lib/x86_64-linux-gnu/libc-2.31.so
2021.03.23 10:19:47.310639 [ 183 ] {} <Fatal> BaseDaemon: 5. abort @ 0x25859 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
2021.03.23 10:19:47.423826 [ 183 ] {} <Fatal> BaseDaemon: 6. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:50: DB::handle_error_code(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0x1193d526 in /workspace/clickhouse
2021.03.23 10:19:47.525133 [ 183 ] {} <Fatal> BaseDaemon: 7. ./obj-x86_64-linux-gnu/../src/Common/Exception.cpp:57: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int, bool) @ 0x1193d621 in /workspace/clickhouse
2021.03.23 10:19:47.902107 [ 183 ] {} <Fatal> BaseDaemon: 8. ./obj-x86_64-linux-gnu/../src/Interpreters/ExpressionActions.cpp:536: DB::ExpressionActions::getSmallestColumn(DB::NamesAndTypesList const&) @ 0x1aaff82c in /workspace/clickhouse
2021.03.23 10:19:48.347551 [ 183 ] {} <Fatal> BaseDaemon: 9. ./obj-x86_64-linux-gnu/../src/Interpreters/TreeRewriter.cpp:658: DB::TreeRewriterResult::collectUsedColumns(std::__1::shared_ptr<DB::IAST> const&, bool) @ 0x1b3e0232 in /workspace/clickhouse
2021.03.23 10:19:48.810807 [ 183 ] {} <Fatal> BaseDaemon: 10. ./obj-x86_64-linux-gnu/../src/Interpreters/TreeRewriter.cpp:848: DB::TreeRewriter::analyzeSelect(std::__1::shared_ptr<DB::IAST>&, DB::TreeRewriterResult&&, DB::SelectQueryOptions const&, std::__1::vector<DB::TableWithColumnNamesAndTypes, std::__1::allocator<DB::TableWithColumnNamesAndTypes> > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::TableJoin>) const @ 0x1b3e1e46 in /workspace/clickhouse
2021.03.23 10:19:49.795282 [ 183 ] {} <Fatal> BaseDaemon: 11. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:379: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&)::$_2::operator()(bool) const @ 0x1ac6c23f in /workspace/clickhouse
2021.03.23 10:19:50.806619 [ 183 ] {} <Fatal> BaseDaemon: 12. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:478: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>, std::__1::shared_ptr<DB::IStorage> const&, DB::SelectQueryOptions const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&) @ 0x1ac6a487 in /workspace/clickhouse
2021.03.23 10:19:51.787799 [ 183 ] {} <Fatal> BaseDaemon: 13. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:169: DB::InterpreterSelectQuery::InterpreterSelectQuery(std::__1::shared_ptr<DB::IAST> const&, DB::Context const&, std::__1::shared_ptr<DB::IBlockInputStream> const&, DB::SelectQueryOptions const&) @ 0x1ac6b5eb in /workspace/clickhouse
2021.03.23 10:19:52.368941 [ 183 ] {} <Fatal> BaseDaemon: 14. ./obj-x86_64-linux-gnu/../src/Storages/StorageMerge.cpp:315: DB::StorageMerge::createSources(std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::QueryProcessingStage::Enum const&, unsigned long, DB::Block const&, std::__1::tuple<std::__1::shared_ptr<DB::IStorage>, std::__1::shared_ptr<DB::RWLockImpl::LockHolderImpl>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, std::__1::shared_ptr<DB::Context> const&, unsigned long, bool, bool) @ 0x1b763473 in /workspace/clickhouse
2021.03.23 10:19:52.982034 [ 183 ] {} <Fatal> BaseDaemon: 15. ./obj-x86_64-linux-gnu/../src/Storages/StorageMerge.cpp:227: DB::StorageMerge::read(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0x1b761c9d in /workspace/clickhouse
2021.03.23 10:19:53.455754 [ 183 ] {} <Fatal> BaseDaemon: 16. ./obj-x86_64-linux-gnu/../src/Storages/IStorage.cpp:108: DB::IStorage::read(DB::QueryPlan&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&, std::__1::shared_ptr<DB::StorageInMemoryMetadata const> const&, DB::SelectQueryInfo&, DB::Context const&, DB::QueryProcessingStage::Enum, unsigned long, unsigned int) @ 0x1b68c3cf in /workspace/clickhouse
2021.03.23 10:19:54.401942 [ 183 ] {} <Fatal> BaseDaemon: 17. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:1729: DB::InterpreterSelectQuery::executeFetchColumns(DB::QueryProcessingStage::Enum, DB::QueryPlan&) @ 0x1ac769cf in /workspace/clickhouse
2021.03.23 10:19:55.370333 [ 183 ] {} <Fatal> BaseDaemon: 18. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:967: DB::InterpreterSelectQuery::executeImpl(DB::QueryPlan&, std::__1::shared_ptr<DB::IBlockInputStream> const&, std::__1::optional<DB::Pipe>) @ 0x1ac6f41e in /workspace/clickhouse
2021.03.23 10:19:56.333688 [ 183 ] {} <Fatal> BaseDaemon: 19. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectQuery.cpp:541: DB::InterpreterSelectQuery::buildQueryPlan(DB::QueryPlan&) @ 0x1ac6e456 in /workspace/clickhouse
2021.03.23 10:19:56.750529 [ 183 ] {} <Fatal> BaseDaemon: 20. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:244: DB::InterpreterSelectWithUnionQuery::buildQueryPlan(DB::QueryPlan&) @ 0x1b22d42a in /workspace/clickhouse
2021.03.23 10:19:57.171618 [ 183 ] {} <Fatal> BaseDaemon: 21. ./obj-x86_64-linux-gnu/../src/Interpreters/InterpreterSelectWithUnionQuery.cpp:298: DB::InterpreterSelectWithUnionQuery::execute() @ 0x1b22db9d in /workspace/clickhouse
2021.03.23 10:19:57.713286 [ 183 ] {} <Fatal> BaseDaemon: 22. ./obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:550: DB::executeQueryImpl(char const*, char const*, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool, DB::ReadBuffer*) @ 0x1b4523e1 in /workspace/clickhouse
2021.03.23 10:19:58.282346 [ 183 ] {} <Fatal> BaseDaemon: 23. ./obj-x86_64-linux-gnu/../src/Interpreters/executeQuery.cpp:908: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, DB::Context&, bool, DB::QueryProcessingStage::Enum, bool) @ 0x1b4508aa in /workspace/clickhouse
2021.03.23 10:19:58.857436 [ 183 ] {} <Fatal> BaseDaemon: 24. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:289: DB::TCPHandler::runImpl() @ 0x1be33912 in /workspace/clickhouse
2021.03.23 10:19:59.463166 [ 183 ] {} <Fatal> BaseDaemon: 25. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1520: DB::TCPHandler::run() @ 0x1be3ff28 in /workspace/clickhouse
2021.03.23 10:19:59.517815 [ 183 ] {} <Fatal> BaseDaemon: 26. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:43: Poco::Net::TCPServerConnection::start() @ 0x1f8ea4bc in /workspace/clickhouse
2021.03.23 10:19:59.604040 [ 183 ] {} <Fatal> BaseDaemon: 27. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerDispatcher.cpp:113: Poco::Net::TCPServerDispatcher::run() @ 0x1f8eacc0 in /workspace/clickhouse
2021.03.23 10:19:59.683553 [ 183 ] {} <Fatal> BaseDaemon: 28. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/ThreadPool.cpp:199: Poco::PooledThread::run() @ 0x1fa27463 in /workspace/clickhouse
2021.03.23 10:19:59.757101 [ 183 ] {} <Fatal> BaseDaemon: 29. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread.cpp:56: Poco::(anonymous namespace)::RunnableHolder::run() @ 0x1fa242ed in /workspace/clickhouse
2021.03.23 10:19:59.829530 [ 183 ] {} <Fatal> BaseDaemon: 30. ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread_POSIX.cpp:345: Poco::ThreadImpl::runnableEntry(void*) @ 0x1fa23178 in /workspace/clickhouse
2021.03.23 10:19:59.829941 [ 183 ] {} <Fatal> BaseDaemon: 31. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
2021.03.23 10:19:59.830204 [ 183 ] {} <Fatal> BaseDaemon: 32. clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
2021.03.23 10:20:00.791047 [ 183 ] {} <Fatal> BaseDaemon: Calculated checksum of the binary: 92D58C250A42DBCF06D0FB8F6646B1BD. There is no information about the reference checksum.
2021.03.23 10:20:08.859469 [ 38 ] {} <Fatal> Application: Child process was terminated by signal 6.
```
https://clickhouse-test-reports.s3.yandex.net/22015/c18018ae06f3708ee73d7dba16556be1e18819c7/fuzzer_debug/report.html#fail1

**Comment by alexey-milovidov** [comment]
`SELECT NULL FROM merge(test, '^hits$') PREWHERE _table = 'abc'`

PREWHERE that filters all the tables out with an expression on `_table` virtual column.

**Comment by alexey-milovidov** [comment]
Fixed in #25801.
